service/src/LevelZero.cpp

```diff
@@ -392,6 +458,45 @@ namespace geopm
         return m_devices.at(l0_device_idx).subdevice.engine_domain.at(l0_domain).size();
     }
 
+    int LevelZeroImp::performance_domain_count(int geopm_domain, unsigned int l0_device_idx, int l0_domain) const
```

Comment:
I'm seeing two functions hidden within this one, where their respective interfaces are not 100% aligned, resulting in an interface that could cause confusion. I would recommend breaking this out:
- ``int perf_domain_count_gpu(unsigned int l0_device_idx)``
- ``int perf_domain_count_gpu_chip(unsigned int l0_device_idx, l0_domain)``

This would avoid unnecessarily passing the ``l0_domain`` parameter when calling the sub device variant. It would also get rid of the ``geopm_domain`` parameter, which would avoid having to deal with the case of an invalid domain being passed, currently being handled by returning 0 which could lead to a "silent bug".